Translate Context (#78)
reactjs/es.reactjs.org: content/docs/context.md

  }
  render() {
    let value = this.context;
    /* renderiza algo basado en el valor de MyContext */
  }
}
MyClass.contextType = MyContext;
```

A la propiedad `contextType` en una clase se le puede asignar un objeto Context creado por [`React.createContext()`](#reactcreatecontext). Esto te permite consumir el valor actual más cercano de ese Context utilizando `this.context`. Puedes hacer referencia a esto en cualquiera de los métodos del ciclo de vida, incluida la función de renderizado.

> Note:
>
> Solo puedes suscribirte a un solo Context usando esta API. Si necesitas leer más de una, lee [Consuming Multiple Contexts](#consuming-multiple-contexts).
>

> > Si está utilizando la [sintaxis experimental de campos de clase pública](https://babeljs.io/docs/plugins/transform-class-properties/), puedes usar un campo de clase **static** para inicializar tu `contextType`.

Reviewer: ```suggestion
> Si estás utilizando la [sintaxis experimental de campos de clase pública](https://babeljs.io/docs/plugins/transform-class-properties/), puedes usar un campo de clase **static** para inicializar tu `contextType`.
```

**app.js**
`embed:context/theme-detailed-app.js`

> ### Actualización Context de un componente anidado {#updating-context-from-a-nested-component}

Reviewer: ```suggestion
### Actualizando Context desde un componente anidado {#updating-context-from-a-nested-component}
```

**app.js**
`embed:context/updating-nested-context-app.js`

### Consumir múltiples Context {#consuming-multiple-contexts}

Para mantener el renderizado de Context rápido, React necesita hacer que cada consumidor de contexto sea un nodo separado en el árbol.

`embed:context/multiple-contexts.js`

Si dos o más valores de contexto se usan a menudo juntos, es posible que desees considerar la creación de tu propio componente de procesamiento que proporcione ambos.

## Advertencias {#caveats}

> Debido a que `Context` usa la identidad de referencia para determinar cuándo se debe volver a renderizar, hay algunos errores que podrían provocar renderizados involuntarios en los consumidores cuando se vuelve a renderizar en el padre del proveedor. Por ejemplo, el código a continuación volverá a representar a todos los consumidores cada vez que el Proveedor vuelva a renderizar porque siempre se crea un nuevo objeto para `value`:

Reviewer: ```suggestion
Debido a que `Context` usa la identidad por referencia para determinar cuándo se debe volver a renderizar, hay algunos errores que podrían provocar renderizados involuntarios en los consumidores cuando se vuelve a renderizar en el padre del proveedor. Por ejemplo, el código a continuación volverá a renderizar a todos los consumidores cada vez que el `Proveedor` se vuelva a renderizar porque siempre se crea un nuevo objeto para `value`:
```

`embed:context/reference-caveats-problem.js`

Para evitar esto, levanta el valor al estado del padre:

`embed:context/reference-caveats-solution.js`

## API antigua {#legacy-api}

> Nota
> 

> > React previamente liberado con el API experimental de Context. La antigua API será compatible con todas las versiones 16.x, pero las aplicaciones que la utilicen deberían migrar a la nueva versión. La API antigua se eliminará en una importante futura versión de React. Leer la [documentación antigua de Context aquí](/docs/legacy-context.html).

Reviewer: ```suggestion
> React previamente había liberado el API experimental de Context. La antigua API será compatible con todas las versiones 16.x, pero las aplicaciones que la utilicen deberían migrar a la nueva versión. La API antigua se eliminará en una futura versión importante de React. Lee la [documentación antigua de Context aquí](/docs/legacy-context.html).
```
